src/tsd/PutDataPointRpc.java

```diff
@@ -103,8 +104,11 @@ public void execute(final TSDB tsdb, final HttpQuery query)
     
     final boolean show_details = query.hasQueryStringParam("details");
     final boolean show_summary = query.hasQueryStringParam("summary");
+    final boolean synchronous_write = query.hasQueryStringParam("sync");
     final ArrayList<HashMap<String, Object>> details = show_details
       ? new ArrayList<HashMap<String, Object>>() : null;
+    final HashMap<IncomingDataPoint,Deferred<Object>> deferred_results
+      = new HashMap<IncomingDataPoint,Deferred<Object>>();
```

Comment:
Size the deferred map based on the # of DPs here and only initialize if the "sync" flag has been passed in.